patrick-lab/server.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Client = require('./model/pages.js');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = net.createServer();
+const PORT = process.env.PORT || 3000;
+
+const pool = [];
+
+ee.on('default', (client, string) => {
+  client.socket.write(`Not a valid command: ${string.split(' ', 1)}\n`);
+});
+
+ee.on('/all', (client, string) => {
+  pool.forEach(c => c.socket.write(`${client.nickname}: ${string}`));
+});
+
+ee.on('/nick', (client, string) => {
+  client.nickname = string.trim();
+});
+
+ee.on('/dm', (client, string) => {
+  let target = string.split(' ')[0];
+  let message = string.split(' ').slice(1).join('');
+  pool.forEach(c => {
+    if(c.nickname === target) {
+      c.socket.write(`${client.nickname}: ${message}`);
+    }
+  });
+});
+
+
+server.on('connection', socket => {
+  let client = new Client(socket);
+  pool.push(client);
+  console.log('Current Pool', pool);
+  pool.forEach(c => c.socket.write(`${client.userName} has connected! Nickname: ${client.nickname}\n`));
+
+  socket.on('data', data => {
+    let command = data.toString().split(' ').shift().trim();
+    if(command.startsWith('/all')) {
```

Comment:
No points lost for this, but you could refactor this entire if statement to be:

```
if(command.startsWith('/')) {
      ee.emit(command, client, data.toString().split(' ').slice(1).join(' '));
      return;
    }
```
And things will work the same.